### ISSUE TITLE
vite@4.0.1 + @solidjs/router results in multiple solid instances warning + broken routing in dev mode

### ISSUE THREAD

**Issue by kavsingh** [solidjs/vite-plugin-solid#78]
Hi! Not sure if this is the right place to be asking:

When using
- `vite@4.0.1`
- `vite-plugin-solid@2.5.0`
- `solid-js@1.6.5`
- `@solidjs/router@0.5.1`

in dev mode, routing is broken and the following warnings are observed in console:
```
You appear to have multiple instances of Solid. This can lead to unexpected behavior.
```

Routing works fine in production builds. Have only observed these dev mode issues in when using `vite@4.0.1` - `vite@4.0.0` works fine.

I have a reproduction repo here: https://github.com/kavsingh/solid-router-vite-401

please let me know if this is the wrong place to ask about this :)

thanks!

**Comment by kavsingh** [comment]
This appears to be resolved with the release of [`vite@4.0.2`](https://github.com/vitejs/vite/compare/v4.0.1...v4.0.2)

thanks!

**Comment by beilo** [comment]
thank you
